RELEASE.md

```diff
@@ -5,29 +5,53 @@ There are two types of Renku releases: planned and unplanned.
 A release consists of: 
 
 * Helm chart changes to reflect the new versions of individual components
+* Updates to the documentation and the integration tests (both `cypress` and `acceptance`)
 * Changes to `CHANGELOG.rst`
 * A new tag in this repository, which results in packaging and pushing out new helm charts and corresponding docker images
 
 This procedure should be followed for *any* release:
 
-* Create a release branch (e.g. `0.46.x`), if one does not already exist, with the [release action](https://github.com/SwissDataScienceCenter/renku/actions/workflows/create-release-branch.yml).
-* Create a `CHANGELOG` entry for the release and open a PR; create a deployment, this is the reference for the release.
-* Note that any PR that should go into the release needs to target the release branch _not_ `master`.
-* All release branches should be protected.
+* Create a release branch (e.g. `0.46.x`), if one does not already exist, with the [release action](https://github.com/SwissDataScienceCenter/renku/actions/workflows/create-release-branch.yml). Mind that release branches are protected.
+* Merge new PRs into the release branch. Neither bugfixes nor features should be merged directly into `master`.
+  * Mind that the PRs pointing to a release branch cannot be merged if they break the platform, even temporarily. Be sure all tests pass by using the `/deploy` string without referencing other components or setting extra values; those should be included in the helm chart folder.
+  * There must be no PRs pointing to a release branch when releasing. If a change won't make it to that release, please point it to the following one (or `master` until it targets another release branch).

```

Comment:
It took me a bit to understand that the first point refers to merging into the release branch but the second one is about merging the release branch into master, I think because it's nested under a point that talks about merging PRs into the release branch. Because the release itself is also a PR, just talking about release branches and feature branches and using that wording consistently makes it easier to understand? And if we can keep it short there's a higher chance of people actually reading it.

I'm thinking like
```
* Create a release branch (e.g. `0.46.x`), if one does not already exist, with the [release action](https://github.com/SwissDataScienceCenter/renku/actions/workflows/create-release-branch.yml). # I don't think we need to mention that they are protected, it's just a fact of life
* create feature branches off of the release branch for new features.
  * a feature branch must contain all the changes needed for that feature.
* squash merge feature branches into the release branch.
  * do not merge feature branches if they break the platform, even temporarily.
  * make sure all tests pass with the `/deploy` action.
  * make sure the `/deploy` action does not reference specific component versions, these should be set in the chart.
  * make sure the `CHANGELOG.rst` is up to date and reflects the changes made.
* The release manager merges the release branch into master when planned
  * If a feature branch won't make it into a planned release, point it to the next release branch.
  * This might need a final cleanup PR to consolidate the changelog
```

Reply:
Thanks! That reads much better.
I slightly changed the 3rd bullet point to make it more readable. Feel free to suggest further changes